Core/src/org/sleuthkit/autopsy/logicalimager/dsp/AddLogicalImageTask.java

```diff
@@ -182,70 +274,171 @@ private String addReport(Path reportPath, String reportName) {
-                LOGGER.log(Level.SEVERE, "Unable to index blackboard artifact " + artifact.getArtifactID(), ex); //NON-NLS
+            artifacts.add(artifact);
+        }
+    }
+
+    private void addExtractedFiles(File src, Path resultsPath, List<Content> newDataSources) throws TskCoreException, UnsupportedEncodingException, IOException {
+
+        SleuthkitCase skCase = Case.getCurrentCase().getSleuthkitCase();
+        SleuthkitCase.CaseDbTransaction trans = null;
+        try {
+            trans = skCase.beginTransaction();
+            LocalFilesDataSource localFilesDataSource = skCase.addLocalFilesDataSource(deviceId, this.src.getName(), timeZone, trans);
+            LocalFileImporter fileImporter = new LocalFileImporter(skCase, trans);
+
+            try (BufferedReader br = new BufferedReader(new InputStreamReader(
+                    new FileInputStream(resultsPath.toFile()), "UTF8"))) { // NON-NLS
+                String line;
+                br.readLine(); // skip the header line
+                int lineNumber = 2;
+                while ((line = br.readLine()) != null) {
+                    String[] fields = line.split("\t", -1); // NON-NLS
+                    if (fields.length != 14) {
+                        throw new IOException(Bundle.AddLogicalImageTask_notEnoughFields(lineNumber, fields.length, 14));
+                    }
+                    String vhdFilename = fields[0];
+//                String fileSystemOffsetStr = fields[1];
+//                    String fileMetaAddressStr = fields[2];
+//                String extractStatusStr = fields[3];
+//                    String ruleSetName = fields[4];
+//                    String ruleName = fields[5];
+//                String description = fields[6];
+                    String filename = fields[7];
+                    String parentPath = fields[8];
+                    String extractedFilePath = fields[9];
+                    String crtime = fields[10];
+                    String mtime = fields[11];
+                    String atime = fields[12];
+                    String ctime = fields[13];
+                    parentPath = ROOT_STR + "/" + vhdFilename + "/" + parentPath;
+
+                    //addLocalFile here 
+                    fileImporter.addLocalFile(
+                            Paths.get(src.toString(), extractedFilePath).toFile(),
+                            filename, 
+                            parentPath,
+                            Long.parseLong(ctime),
+                            Long.parseLong(crtime),
+                            Long.parseLong(atime),
+                            Long.parseLong(mtime),
+                            localFilesDataSource);
+
+                    lineNumber++;
+                } // end reading file
+
+                trans.commit();
+                newDataSources.add(localFilesDataSource);
+            }
+        } catch (TskCoreException ex) {
```

Comment:
This needs to catch all the possible exceptions so we ensure the transaction gets closed.